core/src/main/java/com/graphhopper/storage/BaseGraph.java

```diff
@@ -1220,6 +1221,26 @@ public EdgeIteratorState setWayGeometry(PointList pillarNodes) {
         public PointList fetchWayGeometry(FetchMode mode) {
             return baseGraph.fetchWayGeometry_(edgePointer, reverse, mode, getBaseNode(), getAdjNode());
         }
+        
+        @Override
+        public boolean intersects(Shape shape) {
+            if (!hasBBox) {
+                NodeAccess nodeAccess = baseGraph.getNodeAccess();
+                towerBBox.update(nodeAccess.getLat(getBaseNode()), nodeAccess.getLon(getBaseNode()));
+                towerBBox.update(nodeAccess.getLat(getAdjNode()),  nodeAccess.getLon(getAdjNode()));
+                hasBBox = true;
+            }
+            
+            if (!towerBBox.intersects(shape.getBounds())) {
+                return false;
+            }
+            
+            if (shape instanceof Polygon && ((Polygon) shape).isRectangle()) {
+                return true;
+            }
+            
+            return shape.intersects(fetchWayGeometry(FetchMode.ALL).makeImmutable());
```

Comment:
We could also cache the way geometry here and drop it when onNext() or setWayGeometry() is called.